tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -123,13 +130,29 @@ private:
    std::vector<RSubFieldInfo> fSubFieldsInfo;
    std::size_t fMaxAlignment = 1;
 
+   /// The staging area stores inputs to I/O rules according to the offsets given by the streamer info of
+   /// "TypeName@@Version". The area is allocated depending on I/O rules resp. the source members of the I/O rules.
+   std::unique_ptr<unsigned char[]> fStagingArea;
+   std::unordered_map<std::string, RStagingItem> fStagingItems; ///< Lookup staging items by member name
+
 private:
    RClassField(std::string_view fieldName, const RClassField &source); ///< Used by CloneImpl
    RClassField(std::string_view fieldName, std::string_view className, TClass *classp);
    void Attach(std::unique_ptr<RFieldBase> child, RSubFieldInfo info);
-   /// Register post-read callbacks corresponding to a list of ROOT I/O customization rules. `classp` is used to
-   /// fill the `TVirtualObject` instance passed to the user function.
-   void AddReadCallbacksFromIORules(const std::span<const TSchemaRule *> rules, TClass *classp = nullptr);
+
+   /// Returns the id of member 'name' in the class field given by 'fieldId', or kInvalidDescriptorId if no such
+   /// member exist. Looks recursively in base classes.
+   DescriptorId_t LookupMember(const RNTupleDescriptor &desc, std::string_view memberName, DescriptorId_t classFieldId);
+   /// If there are rules with inputs (source members), create the staging area according to the TClass instance
+   /// that corresponds to the on-disk field.
+   void PrepareStagingArea(const std::vector<const TSchemaRule *> &rules, const RNTupleDescriptor &desc,
+                           const RFieldDescriptor &classFieldId);
+   /// Register post-read callback corresponding to a ROOT I/O customization rules.
+   void AddReadCallbacksFromIORule(const TSchemaRule *rule);
+   /// Given the on-disk information from the page source, find all the I/O customization rules that apply.
```

Comment:
```suggestion
   /// Given the on-disk information from the page source, find all the I/O customization rules that apply.
   /// Rules applies if the 'field' is considered a 'source' of the rule.
```

Reply:
If I understand the implementation correctly, `fieldDesc` is the descriptor corresponding to the class, not of the rule source fields. In that case the proposed addition is incorrect.

(If the addition is correct, the grammar should be `Rules apply` or `A rule applies`)